File: frontend/src/types/post.ts
import { User } from './auth';

export interface Comment {
  id: number;
  content: string;
  created_at: string;
  updated_at: string;
  author: User;
  post_id: number;
}

export interface Post {
  id: number;
  title: string;
  content: string;
  created_at: string;
  updated_at: string;
  author: User;
  comments: Comment[];
  view_count: number;
}

export interface PostListItem {
  id: number;
  title: string;
  created_at: string;
  author: User;
  comment_count: number;
}

File: backend/app/models/post.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.db.database import Base
from datetime import datetime

class Post(Base):
    __tablename__ = "posts"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    title = Column(String(255), nullable=False)
    content = Column(Text, nullable=False)
    views = Column(Integer, default=0)
    created_at = Column(DateTime, default=datetime.now)
    updated_at = Column(DateTime, default=datetime.now, onupdate=datetime.now)

    author = relationship("User", back_populates="posts")
    comments = relationship("Comment", back_populates="post")
